pkg/autostart/autostart.go

```diff
@@ -0,0 +1,112 @@
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateFile respect host OS arch and create unit file
+func CreateFileAndStart(hostOS, instName, workDir string) error {
+	tmpl, err := renderTemplate(hostOS, workDir, instName, os.Executable)
+	if err != nil {
+		return err
+	}
+	unitPath := GetFilePath(hostOS, instName)
+	if _, err := os.Stat(unitPath); !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+		return err
+	}
+	return startStopService("start", hostOS, GetFilePath(hostOS, instName))
+}
+
+// DeleteFile respect host OS arch and delete unit file
+func DeleteFileAndStop(hostOS, instName string) error {
+	unitPath := GetFilePath(hostOS, instName)
+	if _, err := os.Stat(unitPath); err != nil {
+		// Skip error log if file not present
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		return err
+	}
+	if err := startStopService("stop", hostOS, GetFilePath(hostOS, instName)); err != nil {
+		return err
+	}
+	return os.Remove(unitPath)
+}
+
+// GetFilePath returns the path to autostart file with respect of host
+func GetFilePath(hostOS, instName string) string {
+	var fileTmp string
+	if hostOS == "darwin" { // launchd plist
+		fileTmp = fmt.Sprintf("Library/LaunchAgents/io.lima-vm.autostart.%s.plist", instName)
+	}
+	if hostOS == "linux" { // systemd service
+		// Use instance name as argument to systemd service
+		// Instance name available in unit file as %i
+		fileTmp = fmt.Sprintf(".config/systemd/user/lima-vm@%s.service", instName)
+	}
+	return filepath.Join(os.Getenv("HOME"), fileTmp)
+}
+
+func startStopService(action, hostOS, serviceWithPath string) error {
+	// Get filename without extension
+	filenameExt := path.Base(serviceWithPath)
+	filename := strings.TrimSuffix(filenameExt, filepath.Ext(filenameExt))
+
+	var args []string
+	if hostOS == "darwin" {
+		// man launchctl
+		serviceTarget := fmt.Sprintf("gui/%s/%s", strconv.Itoa(os.Getuid()), filename)
+		if action == "start" {
+			args = append(args, []string{"launchctl", "enable", serviceTarget}...)
+		} else {
+			args = append(args, []string{"launchctl", "disable", serviceTarget}...)
```

Comment:
`enable` / `disable` only enable/disable startup at the next login, they do not start/stop. To start, you need to `enable` and then `bootstrap`. To stop, you need to `bootout` and then `disable`, or use `load -w` / `unload -w`.

If it is disabled, `bootstrap` will fail, so `enable` it first.
```bash
launchctl enable gui/501/io.lima-vm.autostart.docker
launchctl bootstrap gui/501 ~/Library/LaunchAgents/io.lima-vm.autostart.docker.plist
```
The order of `bootout` and `disable` is irrelevant, as they do not interfere.
```bash
launchctl bootout gui/501 ~/Library/LaunchAgents/io.lima-vm.autostart.docker.plist
launchctl disable gui/501/io.lima-vm.autostart.docker
```

Are these combinations used differently in `limactl create`, `limactl start`, `limactl stop` with each `--autostart` flag? It seems to get complicated.
My modest hope is that it will start via `launchctl` when `limactl start` if it is registered in `launchd`.

Reply:
> Are these combinations used differently in limactl create, limactl start, limactl stop with each --autostart flag? 

[The autostart file creation](https://github.com/lima-vm/lima/pull/2151#discussion_r1458576167) happens only at the `limactl create` action and doesn't start right away. The `--autostart` flag works only at the create.

/cc @AkihiroSuda @jandubois wdyt?

Reply:
Yes, `limactl create --autostart` will create the instance and the systemd/launchd unit file but will leave the instance stopped.
`limactl start --autostart` should equate to `limactl create --autostart && limactl start`.

Reply:
If `--autostart` is confusing it can be renamed to something else

Reply:
What are the steps to register an already created instance as autostart?

Reply:
limactl edit --autostart

Reply:
Does `limactl edit --autostart=false` unregister it?

Reply:
Thanks! 🙏🏻